### ISSUE TITLE
S/be able to suppress "Log or rethrow exception" 

### ISSUE THREAD

**Issue by PeterJudge-PSC** [Riverside-Software/sonar-openedge#870]
The "Log or rethrow exception" rule at https://sonar.riverside-software.fr/coding_rules?languages=oe&open=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.LogRethrowException&q=rethrow detects when exception handlers swallow exceptions, without giving a way to suppress these critical messages.

1) Intentional swallowing: Certain use-cases - like loggers - require errors to be swallowed. 
2) Actions taken or messages written that do not use anything of the caught exception. Cases include catching Progress.Lang.Stop exceptions, or catching SysErrors when a FIND is done without a NO-ERROR.

This rule can be worked around using something like `INT64(err).` in the catch block but that's uhglee.

Seen in 2.10

**Comment by gquerret** [comment]
> This rule can be worked around using something like INT64(err). in the catch block but that's uhglee.

Never add code like that to get rid of SonarQube issues :-) Flag them as false positives in SonarQube, they won't appear again (even in PDSOE if you're using connected mode). Then complain (nicely) here (as you did) if the rules report invalid issues, in order to fix them. But no, nothing like `INT64(err)` !

Back to the initial problem, you can use `CATCH uncaught AS xxx` to skip this issue.

**Comment by gquerret** [comment]
And does `S/be` mean anything in the title ? Or is that a typo ?

**Comment by PeterJudge-PSC** [comment]
Lazy typing. S/be == "should be"

Sent from Nine<http://www.9folders.com/>
________________________________

**Comment by PeterJudge-PSC** [comment]
I will do: flag in SQ and try "uncaught"

Sent from Nine<http://www.9folders.com/>
________________________________

**Comment by gquerret** [comment]
Did you try `uncaught` variable name ?

**Comment by PeterJudge-PSC** [comment]
Yes, that works for me with `nowarn(varusage)` like below. 

````
{&_proparse_ prolint-nowarn(varusage)}
catch uncaught as Progress.Lang.Error:
end catch.
````

**Comment by gquerret** [comment]
The preprocessor directive should not be required. What version of the plugins do you use ?

**Comment by PeterJudge-PSC** [comment]
@gquerret  2.10.0
